x-pack/plugins/observability/public/pages/alerts/default_cell_actions.tsx

```diff
@@ -79,31 +59,54 @@ const filterCellActions: TGridCellAction[] = [
   },
 ];
 
-/** actions common to all cells (e.g. copy to clipboard) */
-const commonCellActions: TGridCellAction[] = [
+/** actions for adding filters to the search bar */
+const buildFilterCellActions = (addToQuery: (value: string) => void): TGridCellAction[] => [
   ({ data }: { data: TimelineNonEcsData[][] }) => ({ rowIndex, columnId, Component }) => {
-    const { timelines } = useKibanaServices();
-
     const value = getMappedNonEcsValue({
       data: data[rowIndex],
       fieldName: columnId,
     });
+    const text = useMemo(() => `${columnId}${value != null ? `: "${value}"` : ''}`, [
+      columnId,
+      value,
+    ]);
+    const onClick = useCallback(() => {
+      addToQuery(text);
+    }, [text]);
 
-    return (
-      <>
-        {timelines.getHoverActions().getCopyButton({
-          Component,
-          field: columnId,
-          isHoverAction: false,
-          ownFocus: false,
-          showTooltip: false,
-          value,
-        })}
-      </>
+    return Component ? (
+      <Component
+        aria-label={FILTER_FOR_VALUE}
+        data-test-subj="filter-for-value"
+        iconType="plusInCircle"
+        onClick={onClick}
+        title={FILTER_FOR_VALUE}
+      >
+        {FILTER_FOR_VALUE}
+      </Component>
+    ) : (
+      <EuiButtonIcon
+        aria-label={FILTER_FOR_VALUE}
+        className="timelines__hoverActionButton"
+        data-test-subj="filter-for-value"
+        iconSize="s"
+        iconType="plusInCircle"
+        onClick={onClick}
+      />
     );
   },
 ];
 
 /** returns the default actions shown in `EuiDataGrid` cells */
-export const getDefaultCellActions = ({ enableFilterActions }: { enableFilterActions: boolean }) =>
-  enableFilterActions ? [...filterCellActions, ...commonCellActions] : [...commonCellActions];
+export const getDefaultCellActions = ({
+  enableFilterActions,
```

Comment:
Do we need the `enableFilterActions` flag at all?

Reply:
I can get rid of it.